groups/wck-infrastructure/asg_backend.tf

```diff
@@ -0,0 +1,109 @@
+  kms_key_id        = lookup(each.value, "kms_key_id", local.logs_kms_key_id)
+
+  tags = merge(
+    local.default_tags,
+    map(
+      "ServiceTeam", "${upper(var.application)}-BEP-Support"
+    )
+  )
+}
+
+# ASG Scheduled Shutdown
+resource "aws_autoscaling_schedule" "bep-schedule-stop" {
+  count = var.bep_asg_schedule_stop ? 1 : 0
+
+  scheduled_action_name  = "${var.aws_account}-${var.application}-bep-scheduled-shutdown"
+  min_size               = 0
+  max_size               = 0
+  desired_capacity       = 0
+  recurrence             = "00 20 * * 1-5" #Mon-Fri at 8pm
+  autoscaling_group_name = module.bep_asg.this_autoscaling_group_name
+}
+
+# ASG Scheduled Startup
+resource "aws_autoscaling_schedule" "bep-schedule-start" {
+  count = var.bep_asg_schedule_start ? 1 : 0
+
+  scheduled_action_name  = "${var.aws_account}-${var.application}-bep-scheduled-startup"
+  min_size               = var.bep_asg_min_size
+  max_size               = var.bep_asg_max_size
+  desired_capacity       = var.bep_asg_desired_capacity
+  recurrence             = "00 06 * * 1-5" #Mon-Fri at 6am
+  autoscaling_group_name = module.bep_asg.this_autoscaling_group_name
+}
+
+# ASG Module
+module "bep_asg" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/terraform-aws-autoscaling?ref=tags/1.0.36"
+
+  name = "${var.application}-bep"
+  # Launch configuration
+  lc_name       = "${var.application}-bep-launchconfig"
+  image_id      = data.aws_ami.wck_bep_ami.id
+  instance_type = var.bep_instance_size
+  security_groups = [
+    module.wck_bep_asg_security_group.this_security_group_id,
+    data.aws_security_group.nagios_shared.id
+  ]
+  root_block_device = [
+    {
+      volume_size = "40"
+      volume_type = "gp2"
+      encrypted   = true
+      iops        = 0
+    },
+  ]
+  # Auto scaling group
+  asg_name                       = "${var.application}-bep-asg"
+  vpc_zone_identifier            = data.aws_subnet_ids.application.ids
+  health_check_type              = "ELB"
+  min_size                       = var.bep_asg_min_size
+  max_size                       = var.bep_asg_max_size
+  desired_capacity               = var.bep_asg_desired_capacity
+  health_check_grace_period      = 300
+  wait_for_capacity_timeout      = 0
+  force_delete                   = true
+  enable_instance_refresh        = true
+  refresh_min_healthy_percentage = 50
+  refresh_triggers               = ["launch_configuration"]
+  key_name                       = aws_key_pair.wck_keypair.key_name
+  termination_policies           = ["OldestLaunchConfiguration"]
+  iam_instance_profile           = module.wck_bep_profile.aws_iam_instance_profile.name
+  user_data_base64               = data.template_cloudinit_config.bep_userdata_config.rendered
+
+  tags_as_map = merge(
+    local.default_tags,
+    map(
+      "ServiceTeam", "${upper(var.application)}-BEP-Support"
+    )
+  )
+}
+
```

Comment:
Could you also add the ASG alarms to this?
Example here:
`https://github.com/companieshouse/ceu-terraform/blob/main/groups/ceu-frontend/asg_frontend.tf`